pkg/operator/ceph/csi/spec.go

```diff
@@ -311,5 +316,49 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 			return fmt.Errorf("failed to create rbd service: %+v\n%+v", err, cephfsService)
 		}
 	}
+
+	if ver.Major > KubeMinMajor || (ver.Major == KubeMinMajor && ver.Minor >= provDeploymentSuppVersion) {
+		err = createCSIDriverInfo(clientset, RBDDriverName)
+		if err != nil {
+			return fmt.Errorf("failed to create CSI driver object for %q: %+v", RBDDriverName, err)
+		}
+		err = createCSIDriverInfo(clientset, CephFSDriverName)
+		if err != nil {
+			return fmt.Errorf("failed to create CSI driver object for %q: %+v", CephFSDriverName, err)
+		}
+	}
 	return nil
 }
+
+// createCSIDriverInfo Registers CSI driver by creating a CSIDriver object
+func createCSIDriverInfo(clientset kubernetes.Interface, name string) error {
```

Comment:
what about adding a check to the integration tests that the csi registration is created?

Reply:
if the CSI driver object is not created we won't be able to attach PVC to an app pod. as we are validating mounting PVC to App pod, this will be covered internally.